File: src/backend/couponService/src/models/responses/apiResponseHandler.ts
import { FastifyReply } from "fastify";
import { OkResponse } from "./success";
import { ErrorResponse } from "./error";

export class ApiResponseHandler {
  static sendSuccessResponse<T>(
    reply: FastifyReply,
    statusCode: number,
    result: T
  ): void {
    const response: OkResponse<T> = {
      isSuccess: true,
      statusCode,
      result,
    };
    reply.status(statusCode).send(response);
  }

  static sendErrorResponse(
    reply: FastifyReply,
    statusCode: number,
    error: string
  ): void {
    const response: ErrorResponse = {
      isSuccess: false,
      statusCode,
      error,
    };
    reply.status(statusCode).send(response);
  }
}

File: src/backend/shoppingCartService/Models/Responses/ApiResponseHandler.cs
namespace shoppingCartService.Models.Responses;

public static class ApiResponseHandler
{
    public static OkResponse SendSuccessResponse(int statusCode, object result)
    {
        var responseData = new OkResponse
        {
            IsSuccess = true,
            StatusCode = statusCode,
            Result = result
        };
        return responseData;
    }

    public static ErrorResponse SendErrorResponse(int statusCode, string error)
    {
        var responseData = new ErrorResponse
        {
            IsSuccess = false,
            StatusCode = statusCode,
            Error = error,
        };

        return responseData;
    }
}